project_cost_report/static/src/js/report_action.js

```diff
@@ -0,0 +1,456 @@
+            ["revenue", "=", false],
+        ];
+    },
+    /**
+     * Handle a click on the total of a time category.
+     *
+     * The list of analytic lines contained in the category is opened.
+     *
+     * @param {jQuery.Event} - the click event
+     */
+    hourTotalClicked(event){
+        event.preventDefault();
+        var taskTypeId = getRecordIdFromEvent(event);
+        var taskTypeName = getRecordNameFromEvent(event);
+        var actionName = _t("Analytic Lines ({category})").replace("{category}", taskTypeName);
+        this.do_action({
+            name: actionName,
+            res_model: "account.analytic.line",
+            views: [[false, "list"], [false, "form"]],
+            type: "ir.actions.act_window",
+            domain: this.getTimeCategoryDomain(event),
+        });
+    },
+    /**
+     * Handle a click on the label of an outsourcing category.
+     *
+     * @param {jQuery.Event} - the click event
+     */
+    outsourcingCategoryClicked(event){
+        event.preventDefault();
+        var categoryId = getRecordIdFromEvent(event);
+        this.categoryClicked(categoryId, "outsourcing");
+    },
+    /**
+     * Get the domain of analytic lines for outsourcing.
+     *
+     * @returns {Array} - the domain
+     */
+    getOutsourcingCategoryDomain(){
+        return [
+            ["account_id.project_ids", "=", this.projectId],
+            ["task_id", "=", false],
+            ["product_id.type", "=", "service"],
+            ["revenue", "=", false],
+        ];
+    },
+    /**
+     * Handle a click on the total of an outsourcing category.
+     *
+     * The list of analytic lines contained in the category is opened.
+     *
+     * @param {jQuery.Event} - the click event
+     */
+    outsourcingTotalClicked(event){
+        event.preventDefault();
+        var categoryName = getRecordNameFromEvent(event);
+        var actionName = _t("Analytic Lines ({category})").replace("{category}", categoryName);
+        this.do_action({
+            name: actionName,
+            res_model: "account.analytic.line",
+            views: [[false, "list"], [false, "form"]],
+            type: "ir.actions.act_window",
+            domain: this.getOutsourcingCategoryDomain(),
+        });
+    },
+    /**
+     * Handle a click on the name of a purchase order (POXXXXX)
+     *
+     * The form view of the PO is opened.
+     *
+     * @param {jQuery.Event} - the click event
+     */
+    purchaseOrderNameClicked(event){
+        event.preventDefault();
+        var orderId = getRecordIdFromEvent(event);
+        this.do_action({
+            res_model: "purchase.order",
+            views: [[false, "form"]],
+            type: "ir.actions.act_window",
+            res_id: orderId,
+        });
+    },
+    /**
+     * Show the widget.
+     *
+     * This is a standard method of web.Widget.
+     * It is called when the report is reopened after a click on the breadcrumb.
+     */
+    do_show(){
```

Comment:
Codacy found an issue: [Identifier 'do_show' is not in camel case.](https://app.codacy.com/gh/Numigi/odoo-project-addons/pullRequest?prid=6165869)